src/library/coverart.cpp

```diff
@@ -185,9 +189,47 @@ QDebug operator<<(QDebug dbg, const CoverInfo& info) {
             .arg(coverInfoToString(info));
 }
 
+QDebug operator<<(QDebug dbg, const CoverInfo::LoadedImage::Result& result) {
+    switch (result) {
+    case CoverInfo::LoadedImage::Result::Ok:
+        return dbg << "Ok";
+    case CoverInfo::LoadedImage::Result::NoImage:
+        return dbg << "NoImage";
+    case CoverInfo::LoadedImage::Result::ErrorMetadataWithEmptyTrackLocation:
+        return dbg << "ErrorMetadataWithEmptyTrackLocation";
+    case CoverInfo::LoadedImage::Result::ErrorRelativeFilePathWithEmptyTrackLocation:
+        return dbg << "ErrorRelativeFilePathWithEmptyTrackLocation";
+    case CoverInfo::LoadedImage::Result::ErrorFilePathDoesNotExist:
+        return dbg << "ErrorFilePathDoesNotExist";
+    case CoverInfo::LoadedImage::Result::ErrorLoadingFailed:
+        return dbg << "ErrorLoadingFailed";
+    case CoverInfo::LoadedImage::Result::ErrorUnknown:
+        return dbg << "ErrorUnknown";
+    }
+    DEBUG_ASSERT(!"unreachable");
+    return dbg;
+}
+
+QDebug operator<<(QDebug dbg, const CoverInfo::LoadedImage& loadedImage) {
+    dbg = dbg.maybeSpace() << "CoverInfo::LoadedImage";
+    return dbg.nospace()
+            << '{'
+            << loadedImage.filePath
+            << ','
+            << loadedImage.image.size()
+            << ','
+            << loadedImage.result
+            << '}';
```

Comment:
Please add track info here so we can identify the affected track and unassign the cover image manually. That was the issue previously when I noticed the warning (the xruns actually) in a live session but couldn't resolve it except shutting down and messing with the database.

Reply:
Not available in LoadedImage. I will add it in CoverArtCache.

Reply:
Unfortunately this information is not available in CoverArtCache. Then we need to propagate this information up to BaseCoverArtDelegate.